File: linguagem_dart/encapsulamento/esporte/praticante.dart
class Praticante {
  // Válido para um participante, duplas e times
  String _nome = "";
  int _quantidade = 0;

  // Construtor
  Praticante(String nomeDoPraticante, int quantidadeDeParticipantes){ 
    setNome = nomeDoPraticante; 
    setQuantidade = quantidadeDeParticipantes;
  }

  // Getter e setter de nome
  String get getNome => _nome;

  set setNome(String nomeDoParticipante) {
    _nome = nomeDoParticipante;
  }

  // Getter e setter de quantidade
  int get getQuantidade => _quantidade;

  set setQuantidade(int quantidadeDeParticipantes) {
    _quantidade = quantidadeDeParticipantes;
  }
}

File: linguagem_java/encapsulamento/esporte/Praticante.java
package encapsulamento.esporte;

public class Praticante {
    // Válido para individual, duplas e times
    String nome;
    int quantidade;

    // Construtor da classe
    public Praticante(String nomeDoPraticante, int quantidadeDePraticantes) {
        setNome(nomeDoPraticante);
        setQuantidade(quantidadeDePraticantes);
    }

    // Getter e setter de nome
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    // Getter e setter de modalidade
    public int getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(int quantidade) {
        this.quantidade = quantidade;
    }
}
